gp-advanced-select/gpadvs-enable-globally.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Gravity Perks // Advanced Select // Load Advanced Select Globally
+ * https://gravitywiz.com/documentation/gravity-forms-advanced-select/
+ *
+ * Enable Advanced Select on all supported fields automatically.
+ *
+ * Instructions:
+ *
+ * 1. Install this snippet by following the instructions here:
+ *    https://gravitywiz.com/documentation/how-do-i-install-a-snippet/
+ *
+ * 2. Customize the form_id property at the bottom of this snippet.
+ */
+
+class GPASVS_Enable_Globally {
+
+	public function __construct( $args = array() ) {
+
+		$this->_args = wp_parse_args( $args, array(
+			'form_id' => false,
+		) );
+
+		add_action( 'init', array( $this, 'init' ) );
+	}
+
+	public function is_applicable_form( $form ) {
+
+		$form_id = isset( $form['id'] ) ? $form['id'] : $form;
+
+		return empty( $this->_args['form_id'] ) || (int) $form_id == (int) $this->_args['form_id'];
+	}
+
+	public function init() {
+
+		if ( ! is_callable( 'gp_advanced_select' ) ) {
+			return;
+		}
+
+		add_filter( 'gform_pre_render', array( $this, 'enable_gpadvs_for_all_fields' ) );
+		add_filter( 'gform_pre_validation', array( $this, 'enable_gpadvs_for_all_fields' ) );
```

Comment:
```suggestion
		add_filter( 'gform_pre_validation', array( $this, 'enable_gpadvs_for_all_fields' ) );
		add_action( 'gform_pre_enqueue_scripts', array( $this, 'enable_gpadvs_for_all_fields' ) );
```

I believe this should fix preview as well.

Reply:
Thank you @spivurno. I wasn't aware about `gform_pre_enqueue_scripts`, TIL!